File: examples/android-host-shell/feature-bookmarks/src/main/java/com/doogat/hostshell/bookmarks/BookmarksModule.kt
package com.doogat.hostshell.bookmarks

import com.doogat.hostshell.DDBModule
import uniffi.ddb_core.DoogatDriver

object BookmarksModule : DDBModule {
    override val tables = listOf("category", "bookmark")

    override fun bootstrap(driver: DoogatDriver) {
        driver.executeSql("CREATE TABLE IF NOT EXISTS category (name TEXT NOT NULL)")
        driver.executeSql("""
            CREATE TABLE IF NOT EXISTS bookmark (
                title TEXT NOT NULL,
                url TEXT NOT NULL,
                category TEXT REFERENCES category(id)
            )
        """.trimIndent())
    }
}

File: examples/ios-host-shell/HostShell/Sources/Modules/Bookmarks/BookmarksModule.swift
import Foundation

struct BookmarksModule: DDBModule {
    static let tables = ["category", "bookmark"]

    static func bootstrap(_ driver: DoogatDriver) throws {
        _ = try driver.executeSql(sql: "CREATE TABLE IF NOT EXISTS category (name TEXT NOT NULL)")
        _ = try driver.executeSql(sql: """
            CREATE TABLE IF NOT EXISTS bookmark (
                title TEXT NOT NULL,
                url TEXT NOT NULL,
                category TEXT REFERENCES category(id)
            )
        """)
    }
}
